jobs/audit_jobs.py

```diff
@@ -0,0 +1,112 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Audit jobs that validate all of the storage models in the datastore."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from jobs import base_jobs
+from jobs import jobs_utils
+from jobs.decorators import audit_decorators
+from jobs.transforms import base_model_audits
+
+import apache_beam as beam
+
+# IMPORTANT: DO NOT REMOVE THIS IMPORT! It triggers all of the audit decorators.
+# The audit decorators register which kinds of models a DoFn should apply to,
+# and AuditAllStorageModelsJob uses the registry to programmatically apply them.
+from jobs.transforms import * # isort:skip  pylint: disable=import-only-modules, ungrouped-imports, wildcard-import, wrong-import-position, wrong-import-order
+
+
+class AuditAllStorageModelsJob(base_jobs.JobBase):
+    """Runs a comprehensive audit on every model in storage."""
+
+    def run(self):
+        """Returns a PCollection of audit errors aggregated from all models."""
+        if self.job_options.model_getter is None:
+            raise ValueError('JobOptions.model_getter must not be None')
+
+        existing_models, deleted_models = (
+            self.pipeline
+            | 'Get all models' >> self.job_options.model_getter()
+            | 'Partition by model.deleted' >> beam.Partition(
+                lambda model, unused_num_partitions: int(model.deleted), 2)
```

Comment:
Yes it's guaranteed, and we don't have a choice about the signature of this function. I've changed it to `_` to better communicate that we don't need it and to avoid confusion.